polkadot/zombienet-sdk-tests/tests/functional/spam_statement_distribution_requests.rs

```diff
@@ -0,0 +1,163 @@
+
+	let config = NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			r.with_chain("rococo-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_default_args(vec![
+					// parachain::statement-distribution=trace to find
+					// "Peer already being served, dropping request"
+					("-lparachain=debug,parachain::statement-distribution=trace").into(),
+				])
+				.with_default_resources(|r| {
+					r.with_limit_cpu("2")
+						.with_limit_memory("4G")
+						.with_request_cpu("1")
+						.with_request_memory("2G")
+				})
+				.with_genesis_overrides(json!({
+					"configuration": {
+						"config": {
+							"needed_approvals": 2,
+							"scheduler_params": {
+								"max_validators_per_core": 5
+							}
+						}
+					}
+				}))
+				.with_node(|node| node.with_name("honest-0"))
+				.with_node(|node| node.with_name("honest-1"))
+				.with_node(|node| node.with_name("honest-2"))
+				.with_node(|node| node.with_name("honest-3"))
+				.with_node(|node| {
+					node.with_name("malus")
+						.with_image(
+							std::env::var("MALUS_IMAGE")
+								.unwrap_or("docker.io/paritypr/malus".to_string())
+								.as_str(),
+						)
+						.with_command("malus")
+						.with_subcommand("spam-statement-requests")
+						.with_args(vec![
+							"--alice".into(),
+							"--spam-factor=1000".into(),
+							"--insecure-validator-i-know-what-i-do".into(),
+							"-lMALUS=trace,parachain=debug".into(),
+						])
+				})
+		})
+		.with_parachain(|p| {
+			p.with_id(2000)
+				.with_default_command("undying-collator")
+				.cumulus_based(false)
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-2000"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2001)
+				.with_default_command("undying-collator")
+				.cumulus_based(false)
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-2001"))
+		})
+		.build()
+		.map_err(|e| {
+			let errs = e.into_iter().map(|e| e.to_string()).collect::<Vec<_>>().join(" ");
+			anyhow!("config errs: {errs}")
+		})?;
+
+	let spawn_fn = zombienet_sdk::environment::get_spawn_fn();
+	let network = spawn_fn(config).await?;
+
+	let malus = network.get_node("malus")?;
+	let honest = network.get_node("honest-0")?;
+	let relay_client: OnlineClient<PolkadotConfig> = honest.wait_client().await?;
+	let _malus_client: OnlineClient<PolkadotConfig> = malus.wait_client().await?;
+
+	// Check authority status and peers.
+	malus.assert("node_roles", 4.0).await?;
+	honest.assert("node_roles", 4.0).await?;
+
+	// Ensure parachains are registered.
+	assert_para_throughput(
```

Comment:
This doesn't check if registered, only checks for progress.

Reply:
Do we have an equivalent for `honest: parachain 2000 is registered within 60 seconds` ?

Reply:
We don't have it yet

Reply:
Hi 👋,  I will add a `fn` into zombienet to make easy to query paras.